spec.cue

```diff
@@ -0,0 +1,61 @@
+package spec
+
+version: 2
+sources: [...#Source]
+
+
+#StringInt: string | number
+
+#Partition: {
+	name:        string & !=""
+	data_type:   string & !=""
+	expression?: string
+	vals?: {
+		macro: string & !=""
+		args: [string]: #StringInt
```

Comment:
`[string]: #StringInt`

Creates a generic object with a `string` key and allows the value to be either a `string` or an `int`